src/core/border_router/routing_manager.cpp

```diff
@@ -225,22 +255,17 @@ Error RoutingManager::LoadOrGenerateRandomNat64Prefix(void)
     if (Get<Settings>().Read<Settings::Nat64Prefix>(mLocalNat64Prefix) != kErrorNone ||
         !mLocalNat64Prefix.IsValidNat64())
     {
-        Ip6::NetworkPrefix randomNat64Prefix;
-        constexpr uint8_t  nat64PrefixLength = 96;
+        LogNote("No valid NAT64 prefix found in settings, generating new one from /48 BR ULA prefix");
 
-        LogNote("No valid NAT64 prefix found in settings, generating new one");
-
-        // TODO: generate NAT64 prefix from the /48 BR ULA prefix
-        error = randomNat64Prefix.GenerateRandomUla();
-        if (error != kErrorNone)
+        if (!IsValidBrUlaPrefix(mBrUlaPrefix))
         {
-            LogCrit("Failed to generate random NAT64 prefix");
-            ExitNow();
+            SuccessOrExit(error = LoadOrGenerateRandomBrUlaPrefix());
         }
 
-        mLocalNat64Prefix.Clear();
-        mLocalNat64Prefix.Set(randomNat64Prefix);
-        mLocalNat64Prefix.SetLength(nat64PrefixLength);
+        mLocalNat64Prefix                       = mBrUlaPrefix;
```

Comment:
```suggestion
        SuccessOrExit(error = GetBrUlaPrefix(mLocalNat64Prefix));
```
nit: may use a method to encapsulate this logic, so that we don't need to check and call LoadOrGenerateRandomBrUlaPrefix everytime we need the BR ULA prefix?